modes/vterm/evil-collection-vterm.el

```diff
@@ -223,6 +223,22 @@ But don't allow the cursor to move bellow the last prompt line."
   (when (> (count-words (point) (point-max)) 0)
     (evil-next-line count)))
 
+(evil-define-motion evil-collection-vterm-forward-char
+  (count &optional crosslines noerror)
+  "Move cursor to the right by COUNT characters, bypassing line wraps."
+  :type exclusive
+  (if (get-text-property (1+ (point)) 'vterm-line-wrap)
+      (forward-char 2)
+    (evil-forward-char count crosslines noerror)))
+
+(evil-define-motion evil-collection-vterm-backward-char
+  (count &optional crosslines noerror)
+  "Move cursor to the left by COUNT characters, bypassing line wraps."
+  :type exclusive
+  (if (and (not (bobp)) (get-text-property (1- (point)) 'vterm-line-wrap))
```

Comment:
Can you add a comment on what this is checking for.

Reply:
Thank you for the reply @jojojames,

I got the idea from `vterm--get-beginning-of-line`. Basically `(get-text-property (1- (point)) 'vterm-line-wrap)` tells us whether one character to the left is a vterm's line wrap. The `(not (bobp))` is a special case for the first row and first character of the buffer. Because one character to the left is nothing and `evil-backward-char` would throw `if: Args out of range: 0, 0`.

Please let me know what part of this is interesting to you, and I'll add as a code comment.

Reply:
Up to you, what you wrote here seems pretty good to me.

Reply:
Done